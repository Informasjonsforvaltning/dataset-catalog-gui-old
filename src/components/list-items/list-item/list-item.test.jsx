import React from 'react';
import { shallow } from 'enzyme';
import ListItem from './list-item.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    catalogId: '123',
    lastModified: '04/30/2021'
  };
  wrapper = shallow(<ListItem {...defaultProps} />);
});

test('should render ListItem correctly with no props', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render ListItem correctly "DRAFT" item', () => {
  wrapper.setProps({
    title: 'Test item',
    status: 'DRAFT',
    path: '/dataset-catalogs/123/datasets/1'
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render ListItem correctly "PUBLISH" item', () => {
  wrapper.setProps({
    title: 'Test item',
    status: 'PUBLISH',
    path: '/dataset-catalogs/123/datasets/1'
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render ListItem correctly when missing title', () => {
  wrapper.setProps({
    status: 'DRAFT',
    path: '/dataset-catalogs/123/datasets/1'
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render ListItem correctly with lastModified date', () => {
  wrapper.setProps({
    status: 'DRAFT',
    lastModified: '2020-11-30T10:31:06.681',
    path: '/dataset-catalogs/123/datasets/1'
  });
  expect(wrapper).toMatchSnapshot();
});
