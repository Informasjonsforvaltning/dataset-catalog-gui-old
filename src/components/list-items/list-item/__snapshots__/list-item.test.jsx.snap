// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ListItem correctly "DRAFT" item 1`] = `
<div
  className="fdk-list-item d-flex"
>
  <Link
    className="w-100"
    to="/dataset-catalogs/123/datasets/1"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="w-50 fdk-text-size-small"
      >
        Test item
      </span>
      <div
        className="d-flex justify-content-end w-50"
      >
        <span
          className="d-flex justify-content-left w-25"
        >
          04/30/2021
        </span>
        <div
          className="d-flex justify-content-left w-25"
        >
          <Memo(RegistrationStatus)
            registrationStatus="DRAFT"
          />
        </div>
      </div>
    </div>
  </Link>
</div>
`;

exports[`should render ListItem correctly "PUBLISH" item 1`] = `
<div
  className="fdk-list-item d-flex"
>
  <Link
    className="w-100"
    to="/dataset-catalogs/123/datasets/1"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="w-50 fdk-text-size-small"
      >
        Test item
      </span>
      <div
        className="d-flex justify-content-end w-50"
      >
        <span
          className="d-flex justify-content-left w-25"
        >
          04/30/2021
        </span>
        <div
          className="d-flex justify-content-left w-25"
        >
          <Memo(RegistrationStatus)
            registrationStatus="PUBLISH"
          />
        </div>
      </div>
    </div>
  </Link>
</div>
`;

exports[`should render ListItem correctly when missing title 1`] = `
<div
  className="fdk-list-item d-flex"
>
  <Link
    className="w-100"
    to="/dataset-catalogs/123/datasets/1"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="w-50 fdk-text-size-small fdk-color-neutral-darkest"
      >
        <Memo()
          id="listItems.missingTitle"
        />
      </span>
      <div
        className="d-flex justify-content-end w-50"
      >
        <span
          className="d-flex justify-content-left w-25"
        >
          04/30/2021
        </span>
        <div
          className="d-flex justify-content-left w-25"
        >
          <Memo(RegistrationStatus)
            registrationStatus="DRAFT"
          />
        </div>
      </div>
    </div>
  </Link>
</div>
`;

exports[`should render ListItem correctly with lastModified date 1`] = `
<div
  className="fdk-list-item d-flex"
>
  <Link
    className="w-100"
    to="/dataset-catalogs/123/datasets/1"
  >
    <div
      className="d-flex justify-content-between"
    >
      <span
        className="w-50 fdk-text-size-small fdk-color-neutral-darkest"
      >
        <Memo()
          id="listItems.missingTitle"
        />
      </span>
      <div
        className="d-flex justify-content-end w-50"
      >
        <span
          className="d-flex justify-content-left w-25"
        >
          11/30/2020
        </span>
        <div
          className="d-flex justify-content-left w-25"
        >
          <Memo(RegistrationStatus)
            registrationStatus="DRAFT"
          />
        </div>
      </div>
    </div>
  </Link>
</div>
`;

exports[`should render ListItem correctly with no props 1`] = `""`;
